lib/trino-orc/src/test/java/io/trino/orc/TestSliceDictionaryColumnWriter.java

```diff
@@ -45,17 +54,28 @@ public void testDirectConversion()
                 toIntExact(DEFAULT_MAX_COMPRESSION_BUFFER_SIZE.toBytes()),
                 () -> new StringStatisticsBuilder(toIntExact(DEFAULT_MAX_STRING_STATISTICS_LIMIT.toBytes()), new NoOpBloomFilterBuilder()));
 
-        // a single row group exceeds 2G after direct conversion
-        byte[] value = new byte[megabytes(1)];
-        ThreadLocalRandom.current().nextBytes(value);
-        Block data = RunLengthEncodedBlock.create(VARCHAR, Slices.wrappedBuffer(value), 3000);
+        Slice randomUtf8Slice = createRandomUtf8Slice(VALID_CODEPOINTS, megabytes(1));
+        Block data = RunLengthEncodedBlock.create(VARCHAR, Slices.wrappedBuffer(randomUtf8Slice.byteArray()), 3000);
         writer.beginRowGroup();
         writer.writeBlock(data);
         writer.finishRowGroup();
 
         assertFalse(writer.tryConvertToDirect(megabytes(64)).isPresent());
     }
 
+    public static Slice createRandomUtf8Slice(int[] codePointSet, int lengthInBytes)
+    {
+        int[] codePoints = new int[lengthInBytes];
+        ThreadLocalRandom random = ThreadLocalRandom.current();
+        long totalLength = 0;
+        for (int i = 0; i < codePoints.length && totalLength <= lengthInBytes; i++) {
+            int codePoint = codePointSet[random.nextInt(codePointSet.length)];
+            codePoints[i] = codePoint;
+            totalLength += lengthOfCodePoint(codePoint);
```

Comment:
is it OK to produce slightly more than `lengthInBytes`?
or maybe we should pad the tail with 1-byte characters?

Reply:
I can make it slightly bigger